src/input.rs

```diff
@@ -179,12 +182,14 @@ impl<'a> Input<'a> {
                     if self.err.is_none() {
                         self.term.show_cursor()?;
                         self.term.clear_to_end_of_screen()?;
+                        ctrlc::close_ctrlc_handle(ctrlc_handle);
                         return self.handle_submit();
                     }
                 }
                 Key::Tab => self.handle_tab()?,
                 Key::Escape => {
-                    self.clear()?;
+                    self.term.show_cursor()?;
+                    ctrlc::close_ctrlc_handle(ctrlc_handle);
```

Comment:
I think the better way to do this would be to `impl Drop` on a struct that `ctrlc::show_cursor` returns so that showing the cursor happens automatically and we don't need to remember to explicitly close. https://doc.rust-lang.org/rust-by-example/trait/drop.html

Reply:
I think you'll need to wrap ctrlc_handle in a struct like this:

```rust
pub struct CtrlcHandle(Handle);

impl Drop CtrlcHandle {
  fn drop() {
    close_ctrlc_handle(self.0);
  }
}

Reply:
This sounds what i was looking for. So if the variable is going out of scope the handle is closed automatically.